### DISCUSSION TITLE
Mixed definitions/types in list

### DISCUSSION THREAD

**Discussion by dylankb** [cue-lang/cue#1994] [Q&A] [answered]
I'd like to understand if Cue can support a schema where there's a list of multiple different definitions and/or scalar types together.

For example, one thing I tried was to define a list that took a variadic list of definitions plus on other type.

```
#vals: "a" |  "b" |  "c"

#valueList: [#vals, string]
foo: #valueList
foo: ["a", "other"]
```

This does not work though.
```
#valueList: [...#vals, string]
#bar: #valueList
bar: ["a", "other"]
```

https://cuelang.org/play/?id=Nu4xFu1_bCU#cue@export@cue

**Comment by myitcv** [comment, marked as answer]
> #valueList: [...#vals, string]

The issue here is that the variadic aspect of the list can only appear at the end.

If you turn it round, then it works:

```
#vals: "a" | "b" | "c"
#valueList: [string, ...#vals]
#bar: #valueList
bar:  #bar & ["other", "a", "b"]
```

https://cuelang.org/play/?id=TvH1e5HGMwU#cue@export@cue
